File: Backend/src/main/kotlin/com/backend/is/IS_Backend/repository/UserRepository.java
package com.backend.is.IS_Backend.repository;

import mk.ukim.finki.it.reservengo.model.domain.User;
import org.springframework.data.jpa.repository.JpaRepository;
import org.springframework.stereotype.Repository;

import java.time.LocalDateTime;
import java.util.List;
import java.util.Optional;

@Repository
public interface UserRepository extends JpaRepository<User, Long> {
    Optional<User> findByEmail(String email);
    List<User> findAllByEnabledFalseAndLastActivityDateBefore(LocalDateTime time);
    List<User> findAllByEnabledTrueAndLastActivityDateBefore(LocalDateTime time);
}

File: Backend/src/main/kotlin/com/backend/is/IS_Backend/repository/UserRepository.kt
package com.backend.`is`.IS_Backend.repository

import com.backend.`is`.IS_Backend.model.domain.User
import org.springframework.data.jpa.repository.JpaRepository
import org.springframework.data.jpa.repository.Query
import org.springframework.stereotype.Repository
import java.util.Optional

@Repository
interface UserRepository : JpaRepository<User, Long> {
    fun findByEmail(email: String): Optional<User>

    @Query("select u from User u where u.userRole <> com.backend.is.IS_Backend.enumerations.Roles.ROLE_ADMIN")
    fun findAllNonAdmin(): List<User>
}
